gammapy/estimators/map/ts.py

```diff
@@ -236,10 +236,9 @@ def estimate_kernel(self, dataset):
         model = self.model.copy()
         model.spatial_model.position = geom.center_skydir
 
-        # Creating exposure map with exposure at map center
+        # Creating exposure map with the mean non-null exposure
         exposure = Map.from_geom(geom, unit="cm2 s1")
-        exposure_center = dataset.exposure.to_region_nd_map(geom.center_skydir)
-        exposure.data[...] = exposure_center.data
+        exposure.data[...] = np.mean(dataset.exposure.data[dataset.exposure.data > 0.])
```

Comment:
you could also consider the dataset.mask and only finite values to compute the mean exposure :
```python
exposure_mask = (dataset.mask.data) & (dataset.exposure.data > 0.) & np.isfinite(dataset.exposure.data)
exposure.data[...] = np.mean(dataset.exposure.data[exposure_mask])
```

Reply:
this also address my comment about unit

Reply:
I think we can use  dataset.mask.interp_to_geom( dataset.exposure.geom)  as a the mask in reco energy is dilated compared to the mask in true energy we don't loose information .

Reply:
This can not work as the energy axis names are different between the exposure (in true energy) and the mask (in reco energy)... No turn-around found yet.